### PR TITLE
[DOCS] Clarify configuring indices/data streams used by stack monitoring

### PR DESCRIPTION
**Pull request by abdonpijpelink**
The [documentation that discusses configuring the indices used by Stack Monitoring](https://www.elastic.co/guide/en/elasticsearch/reference/master/config-monitoring-indices.html) is incomplete. The steps to configure these monitoring indices or data streams depend on the monitoring method. 

This PR tries to clarify these docs.

Related: 
https://github.com/elastic/elasticsearch/issues/85873
https://github.com/elastic/kibana/issues/153526

### CODE REVIEW

File: docs/reference/monitoring/indices.asciidoc [hunk 1]
```diff
@@ -1,79 +1,143 @@
...
-== Configuring indices for monitoring
+== Configuring data streams/indices for monitoring
 
-Sometimes the default index settings might not work for your situation. For
-example, you might want to change index lifecycle management (ILM) settings,
-add custom mappings, or set the number of shards and replicas
-for the monitoring indices.
+Monitoring data is stored in data streams or indices in {es}. The default data
+stream or index settings may not work for your situation. For example, you might
+want to change index lifecycle management (ILM) settings, mappings, or the
```
Comment (lucabelluccini) [inline docs/reference/monitoring/indices.asciidoc:7]: I would not recommend at all to modify the mappings. Only index settings and ILM policies.

File: docs/reference/monitoring/indices.asciidoc [hunk 2]
```diff
@@ -1,79 +1,143 @@
...
 
-You can change the default behavior. The steps you follow depend on which data
-collection method is used.
+* <<config-monitoring-data-streams-elastic-agent>>
+* <<config-monitoring-data-streams-metricbeat-8>>
+* <<config-monitoring-indices-metricbeat-7-internal-collection>> (the default 
+for the {ess} on {ecloud})
 
-[float]
+IMPORTANT: Changing mappings or settings can cause your monitoring dashboards to
```
Comment (lucabelluccini) [inline docs/reference/monitoring/indices.asciidoc:16]: 👍

File: docs/reference/monitoring/indices.asciidoc [hunk 3]
```diff
@@ -1,79 +1,143 @@
...
+`.monitoring-es-8-mb`. 
+
+The settings and mappings for these data streams are determined by an index
+template named `.monitoring-{product}-mb`. For example: `.monitoring-es-mb`. You
+can alter the settings of each data stream by cloning this index template and
+editing it. 
 
-[float]
-=== Configuring indices created by {metricbeat}
+WARNING: Any future changes to the default index template will not automatically
```
Comment (lucabelluccini) [inline docs/reference/monitoring/indices.asciidoc:63]: I would rephrase to something like: "if you're upgrading the stack, you will likely need to repeat this procedure to get the latest updates of the default index template of Stack Monitoring"

File: docs/reference/monitoring/indices.asciidoc [hunk 4]
```diff
@@ -1,79 +1,143 @@
...
-store the monitoring data that {metricbeat} collects from a cluster.
+You can clone index templates in {kib}:
 
-You can retrieve the templates through the `_template` API:
+* Navigate to *Stack Management* > *Index Management* > *Index Templates*.
+* From the *View* dropdown, select *System templates*.
+* Search for the index template.
+* Select the *Clone* action.
+* Change the name, for example into `custom_monitoring`.
+* Set the priority to `1`, to ensure it overrides the default index template.
```
Comment (lucabelluccini) [inline docs/reference/monitoring/indices.asciidoc:73]: Afaik, the new index templates work in the opposite way:
> If a new data stream or index matches more than one index template, the index template with the highest priority is used.
From https://www.elastic.co/guide/en/elasticsearch/reference/current/index-templates.html

So the user should set the priority > 200

File: docs/reference/monitoring/indices.asciidoc [hunk 5]
```diff
@@ -1,79 +1,143 @@
...
+* Specify the settings you want to change in the `settings` section.
+* Save the cloned template.
+
+You can also use the {es} API:
+
+* Retrieve the index template using the <<indices-get-template,get index 
+template API>>.
+* Edit the index template: set the template `priority` to `1`, and specify the
```
Comment (lucabelluccini) [inline docs/reference/monitoring/indices.asciidoc:81]: Same as above.

File: docs/reference/monitoring/indices.asciidoc [hunk 6]
```diff
@@ -1,79 +1,143 @@
...
-You can retrieve the templates through the `_template` API:
+* Navigate to *Stack Management* > *Index Management* > *Index Templates*.
+* From the *View* dropdown, select *System templates*.
+* Search for the index template.
+* Select the *Clone* action.
+* Change the name, for example into `custom_monitoring`.
+* Set the priority to `1`, to ensure it overrides the default index template.
+* Specify the settings you want to change in the `settings` section.
+* Save the cloned template.
+
```
Comment (lucabelluccini) [inline docs/reference/monitoring/indices.asciidoc:76]: Can we add a warn/info box where we highlight the  use of the new index templates vs the legacy index templates?

**Review by lucabelluccini — CHANGES_REQUESTED** [review]

File: docs/reference/monitoring/indices.asciidoc [hunk 7]
```diff
@@ -1,79 +1,143 @@
...
-while giving you a single named resource for requests. You can alter the
-settings of each dataset by configuring an `@custom` component template.
-For more information, refer to {fleet-guide}/data-streams.html[Data streams].
+When <<configuring-elastic-agent,monitoring using {agent}>>, data is stored in a
+set of data streams named `metrics-{product}.stack_monitoring.{dataset}`. For
+example: `metrics-elasticsearch.stack_monitoring.shard-default`. 
+
+The settings and mappings for these data streams are determined by an index
+template named `metrics-{product}.stack_monitoring.{dataset}`. For example:
+`metrics-elasticsearch.stack_monitoring.shard`.
```
Comment (klacabane) [inline docs/reference/monitoring/indices.asciidoc:28]: We can add the trailing namespace as documented in the [data stream naming scheme](https://www.elastic.co/blog/an-introduction-to-the-elastic-data-stream-naming-scheme)

```suggestion
template named `metrics-{product}.stack_monitoring.{dataset}-{namespace}`. For example:
`metrics-elasticsearch.stack_monitoring.shard-default`. 
```

File: docs/reference/monitoring/indices.asciidoc (hunk 1 above)
Comment (klacabane) [reply docs/reference/monitoring/indices.asciidoc:7]: I'd leave it the way it was phrased before, _"add custom mappings"_, since that can be legitimate use case and isn't potentially destructive like a modification

File: docs/reference/monitoring/indices.asciidoc (hunk 4 above)
Comment (abdonpijpelink) [reply docs/reference/monitoring/indices.asciidoc:73]: The default index template didn't have a priority set, so it defaults to a priority of 0, so I figured 1 would be high enough. I'll change it into 500, to follow the guidance of the docs that you linked.

File: docs/reference/monitoring/indices.asciidoc (hunk 7 above)
Comment (klacabane) [reply docs/reference/monitoring/indices.asciidoc:28]: Ah you're right, only the data streams have the namespace part, ignore me :)

File: docs/reference/monitoring/indices.asciidoc [hunk 8]
```diff
@@ -1,79 +1,143 @@
...
+
+[[config-monitoring-data-streams-elastic-agent]]
 === Configuring data streams created by {agent}
 
-{agent} uses data streams to store time series data across multiple indices
-while giving you a single named resource for requests. You can alter the
-settings of each dataset by configuring an `@custom` component template.
-For more information, refer to {fleet-guide}/data-streams.html[Data streams].
+When <<configuring-elastic-agent,monitoring using {agent}>>, data is stored in a
+set of data streams named `metrics-{product}.stack_monitoring.{dataset}`. For
```
Comment (abdonpijpelink) [inline docs/reference/monitoring/indices.asciidoc:23]: @klacabane I guess I need to add the `-{namespace}` here.

Comment (klacabane) [reply docs/reference/monitoring/indices.asciidoc:23]: Yes. The namespace can be set when configuring the integration but I don't think it's important to specify here

Screenshot 2023-04-18 at 13 14 07

File: docs/reference/monitoring/indices.asciidoc (hunk 3 above)
Comment (abdonpijpelink) [reply docs/reference/monitoring/indices.asciidoc:63]: I went with: _"You need to repeat this procedure when upgrading the Elastic Stack to get the latest updates to the default monitoring index templates._

**Comment by abdonpijpelink** [comment]
@klacabane @lucabelluccini Thank you for the reviews. I've pushed https://github.com/elastic/elasticsearch/pull/95285/commits/4f7d4e80bed9979898d1793d022cb791fdacf019 to incorporate your feedback.

**Review by klacabane — APPROVED** [review]
Looks great, thank you!

**Comment by klacabane** [comment]
Confirming I've tested the procedures and it works like charm (with the documented limitations).

@lucabelluccini I think your request for change is blocking the merge. Looks like we'll also need a stamp from the docs team

**Review by lucabelluccini — APPROVED** [review]

**Comment by lucabelluccini** [comment]
Thank you @abdonpijpelink & @klacabane for helping with this. 
Really appreciated. Sorry for the delay on approval.

**Comment by abdonpijpelink** [comment]
Great team work. Thanks @klacabane and @lucabelluccini !

**Comment by abdonpijpelink** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|7.17|[](https://github.com/elastic/elasticsearch/pull/95587)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)
